assets/js/googlesitekit/modules/create-info-store.js

```diff
@@ -53,13 +53,40 @@ export const createInfoStore = (
 ) => {
 	invariant( storeName, 'storeName is required.' );
 
+	const WAIT_FOR_REAUTH_RESOLVERS = 'WAIT_FOR_REAUTH_RESOLVERS';
+
 	const initialState = {};
 	const actions = {};
-	const controls = {};
+	const controls = {
+		[ WAIT_FOR_REAUTH_RESOLVERS ]: createRegistryControl(
+			( registry ) => async () => {
+				const { __experimentalResolveSelect } = registry;
+				const { getAuthentication, getConnectURL } =
+					__experimentalResolveSelect( CORE_USER );
+				const { getSiteInfo } =
+					__experimentalResolveSelect( CORE_SITE );
+
+				await Promise.all( [
+					// Authentication is needed for checking `needsReauthentication`.
+					getAuthentication(),
+					// Site info is needed for the `adminURL`.
+					getSiteInfo(),
+					// connectURL is needed for the reAuthURL when reauthentication is needed.
```

Comment:
Very minor but this comment could use some backticks for consistency with the ones above:

```suggestion
					// `connectURL` is needed for the `reAuthURL` when reauthentication is needed.
```